src/Logary.Tests/Registry.fs

```diff
@@ -78,12 +78,69 @@ let tests = [
     do! logm.shutdown ()
   }
 
+  testCaseJob "log with scope" (job {
+    let dataSlot = DataSlot.useDefault ()
+    let! (logm, out, _)  = Utils.buildLogManagerWith (fun r -> r |> Config.middleware (Middleware.fillWithContextSlot dataSlot) )
+
+    for _ in 1..10 do
+      let latch = Latch 4
+      let loggername = PointName.parse "logger.test"
+      let lg1 = logm.getLogger loggername
+
+      let s1 = dataSlot.push (fun _ -> "scope-1")
+      lg1.infoWithBP (eventX "scope1") |> Hopac.startWithActions ignore (latch.Decrement >> start)
+
+      let s2 = dataSlot.push (fun _ -> "scope-2")
+
+      let lg2 = logm.getLogger (PointName.parse "logger.test.another")
+      lg2.infoWithBP (eventX "scope2") |> Hopac.startWithActions ignore (latch.Decrement >> start)
+
+      do s2.Dispose ()
+      lg2.infoWithBP (eventX "2dispose") |> Hopac.startWithActions ignore (latch.Decrement >> start)
+
+      do s1.Dispose ()
+      lg2.infoWithBP (eventX "1dispose") |> Hopac.startWithActions ignore (latch.Decrement >> start)
+
+      do! latch |> Latch.await |> Alt.afterJob logm.flushPending
+
+      let outStr = clearStream out
+      Expect.isRegexMatch outStr (new Regex("""scope1.*?_logary.scope => \["scope-1"\]""", RegexOptions.Singleline)) "shoule have scope-1 value"
+      Expect.isRegexMatch outStr (new Regex("""scope2.*?_logary.scope => \["scope-2", "scope-1"\]""", RegexOptions.Singleline)) "shoule have scope-1/2 value"
+      Expect.isRegexMatch outStr (new Regex("""2dispose.*?_logary.scope => \["scope-1"\]""", RegexOptions.Singleline)) "shoule only have scope-1 value"
+      Expect.isRegexMatch outStr (new Regex("1dispose.*?_logary.scope => \[\]", RegexOptions.Singleline)) "shoule have no scope value"
+
+    do! logm.shutdown ()
+  })
+
+  testCaseJob "log with span" (job {
+    let! (logm, out, _)  = Utils.buildLogManager ()
+    let checkSpanId spanId = job {
+      do! logm.flushPending ()
+      clearStream out
+      |> Expecto.Flip.Expect.stringContains "should has spanId" spanId
```

Comment:
`have`, not `has`. Good that you're using `Expecto.Flip`, but even better if we change the whole file to use it.